stl/inc/condition_variable

```diff
@@ -223,7 +225,6 @@ public:
             _CSTD xtime _Tgt;
             (void) _To_xtime_10_day_clamped(_Tgt, _Rel_time);
             const int _Res = _Cnd_timedwait(_Mycnd(), _Myptr->_Mymtx(), &_Tgt);
-            _Guard.unlock();

```

Comment:
Is there a particular reason to remove this line instead of changing it to `_Guard_unlocks_before_locking_outer.unlock()`? I think it was intentional that we called (the potentially slow) `_Throw_C_error` outside of any locks.

Reply:
Wanted to make it cleaner, but I agree it is better to keep it as is.